### PR TITLE
Feature tls

### PR DESCRIPTION
**Pull request by d-smith**
This feature adds MTLS to atom feed and the proxy configuration, such that a client certificate must be used to gain access to the atom feed.

### CODE REVIEW

File: cmd/atompubcmd.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	"net/http"
+	"os"
+)
+
+type AtomFeedPubConfig struct {
+	linkhost            string
+	listenerHostAndPort string
+	secure              bool
+	privateKey          string
+	certfificate        string
```
Comment (AlexeiSuleac) [inline cmd/atompubcmd.go]: spelling certificate

File: cmd/atompubcmd.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+		configErr = true
+	}
+
+	config.listenerHostAndPort = os.Getenv("LISTENADDR")
+	if config.listenerHostAndPort == "" {
+		log.Println("Missing LISTENADDR environment variable value")
+		configErr = true
+	}
+
+	//Load the TLS config from the envrionment. If INSECURE_PUBLISHER is present
```
Comment (AlexeiSuleac) [inline cmd/atompubcmd.go]: spelling environment

File: cmd/proxy/testread/client.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/xtracdev/tlsconfig"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+)
+
+func main() {
+	if len(os.Args) < 5 {
+		fmt.Println("Need four args: client key file, client cert file, ca cert file, emdpoint")
```
Comment (AlexeiSuleac) [inline cmd/proxy/testread/client.go]: spelling endpoint

This is the only place where information about the arguments needed is provided. Should this be in a README as well ?

File: cmd/atompubcmd.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+	retrieveHandler, err := atompub.NewEventRetrieveHandler(db)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	r := mux.NewRouter()
+	r.HandleFunc("/notifications/recent", recentHandler)
+	r.HandleFunc("/notifications/{feedid}", archiveHandler)
+	r.HandleFunc("/notifications/{aggregate_id}/{version}", retrieveHandler)
```
Comment (AlexeiSuleac) [inline cmd/atompubcmd.go]: what is the difference between `feedid` and `aggregate_id` I mean why use underscore in second case ? Should naming not be consistent here ? Or some other rules or naming conventions are being followed here?

Comment (d-smith) [reply cmd/atompubcmd.go]: Good questions - feedid is for a particulate atom feed, and aggregate id is for an aggregate that has been published via one of the feed pages. Based on our other conventions I should probably change aggregate_id to aggregateId, and I should change the uri as well to /events/{aggregateId}

**Pull request merged**